test/extensions/filters/http/jwt_authn/verifier_test.cc

```diff
@@ -0,0 +1,338 @@
+  requires:
+    requires_all:
+      requirements:
+      - requires_all:
+)";
+  MessageUtil::loadFromYaml(config, proto_config_);
+  createVerifier();
+
+  auto mock_auth = std::make_unique<MockAuthenticator>();
+  EXPECT_CALL(*mock_auth.get(), doVerify(_, _, _)).Times(0);
+  mock_auths_.push_back(std::move(mock_auth));
+
+  EXPECT_CALL(mock_cb_, onComplete(_)).WillOnce(Invoke([](const Status& status) {
+    ASSERT_EQ(status, Status::Ok);
+  }));
+  auto headers = Http::TestHeaderMapImpl{{":path", "/match"}};
+  context_ = VerifyContext::create(headers, &mock_cb_);
+  verifier_->verify(*context_);
+}
+
+TEST_F(VerifierTest, TestRequiresAll) {
+  MessageUtil::loadFromYaml(RequiresAllConfig, proto_config_);
+
+  for (int i = 0; i < 2; ++i) {
+    auto mock_auth = std::make_unique<MockAuthenticator>();
+    EXPECT_CALL(*mock_auth.get(), doVerify(_, _, _))
+        .WillRepeatedly(Invoke(
+            [](Http::HeaderMap&, std::vector<JwtLocationConstPtr>*,
+               std::function<void(const Status& status)> callback) { callback(Status::Ok); }));
+    EXPECT_CALL(*mock_auth.get(), sanitizePayloadHeaders(_)).Times(1);
+    mock_auths_.push_back(std::move(mock_auth));
+  }
+  createVerifier();
+
+  EXPECT_CALL(mock_cb_, onComplete(_)).WillOnce(Invoke([](const Status& status) {
+    ASSERT_EQ(status, Status::Ok);
+  }));
+  auto headers = Http::TestHeaderMapImpl{
+      {":path",
+       "/requires-all?jwt_a=" + std::string(GoodToken) + "&jwt_b=" + std::string(OtherGoodToken)},
+  };
+  context_ = VerifyContext::create(headers, &mock_cb_);
+  verifier_->verify(*context_);
+}
+
+// test requires all with first token returning bad format
+TEST_F(VerifierTest, TestRequiresAllBadFormat) {
+  MessageUtil::loadFromYaml(RequiresAllConfig, proto_config_);
+  std::vector<std::function<void(const Status& status)>> callbacks;
+  for (int i = 0; i < 2; ++i) {
+    auto mock_auth = std::make_unique<MockAuthenticator>();
+    EXPECT_CALL(*mock_auth.get(), doVerify(_, _, _))
+        .WillRepeatedly(Invoke([&](Http::HeaderMap&, std::vector<JwtLocationConstPtr>*,
+                                   std::function<void(const Status& status)> callback) {
+          callbacks.push_back(std::move(callback));
+        }));
+    EXPECT_CALL(*mock_auth.get(), sanitizePayloadHeaders(_)).Times(1);
+    mock_auths_.push_back(std::move(mock_auth));
+  }
+  createVerifier();
+
+  EXPECT_CALL(mock_cb_, onComplete(_)).WillOnce(Invoke([](const Status& status) {
+    ASSERT_EQ(status, Status::JwtBadFormat);
+  }));
+  auto headers = Http::TestHeaderMapImpl{{":path", "/requires-all?jwt_a=xxx&jwt_b=xxx"}};
+  context_ = VerifyContext::create(headers, &mock_cb_);
+  verifier_->verify(*context_);
+  callbacks[0](Status::JwtBadFormat);
+  // can keep invoking callback
+  callbacks[1](Status::Ok);
+  callbacks[0](Status::Ok);
+  callbacks[1](Status::Ok);
+}
+
+// test requires all with second token returning missing jwt
+TEST_F(VerifierTest, TestRequiresAllMissing) {
```

Comment:
verifier_test is important to cover all cases.   Ilike to split it into following test files

1) provider_verifier_test.cc   to only cover provider_verifier, and all its success, failure, missing cases
2) all_verifier_test.cc to test all_fail,  all_allowed cases
2) group_verifier_test.cc to cover  all, any,  and nested all with any or any with all,  and failed, success, missing

Reply:
added more tests for verifier.